hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from skimage import draw
+from scipy.fftpack import ifft2, fftshift, fft2
+from numpy.linalg import norm
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+from hyperspy._signals.signal2d import Signal2D
+
+
+class HologramImage(Signal2D):
+    """Image subclass for holograms acquired via electron holography."""
+
+    _signal_type = 'hologram'
+
+    def reconstruct_phase(self, reference=None, rec_param=None, show_phase=False, return_param=False,
+                          fresnel_fringe_filter=False, verbose=False):
+        """Reconstruct holography data
+
+        Parameters
+        ----------
+        reference : ndarray, class:`~hyperspy._signals.Image`, `~hyperspy._signals.HologramImage`
+            Vacuum reference hologram.
+        rec_param : tuple
+            Reconstruction parameters in sequence (SBrect(x0, y0, x1, y1), SB size)
+        show_phase : boolean
+            set True to plot phase after the reconstruction
+        return_param : boolean
+            set True to return reconstruction as tuple, useful for batch processing
+        fresnel_fringe_filter : boolean
+            set True to apply an automatic filtering of biprisms fresnel fringe
+        verbose : boolean
+            set True to see details of the reconstruction (i.e. SB selection)
+
+        Returns
+        -------
+        wave : WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+        rec_param : tuple
+            see the description in Parameters (optional, triggered by return_param flag)
+
+        Notes
+        -----
+        The reconstruction parameters assigned interactively if rec_param is not givven
+
+        See Also
+        --------
+        _reconstruct
+
+        """
+
+        (sx, sy) = self.data.shape
+        holo_data = self.data
+
+        # Parsing reference input:
+        if reference is None:  # to reconstruct hologram WITHOUT a reference
+            eh_hw_fft = fftshift(fft2(holo_data))
+        elif isinstance(reference, Signal2D):
+            # to reconstruct hologram WITH a reference
+            ref_data = reference.data
+            eh_hw_fft = fftshift((fft2(ref_data)))
+        else:
+            ref_data = reference
+            eh_hw_fft = fftshift((fft2(ref_data)))
+
+        # Preparing reconstruction parameters:
+        if rec_param is None:
+            eh_hw_fft_half = eh_hw_fft[0:np.int(sx / 2.02), :]  # selects half of the fft excluding fraction of points
+            #  in the middle
+            yr, xc = np.unravel_index(eh_hw_fft_half.argmax(), eh_hw_fft_half.shape)  # Center of selected sideBand
+            sb_pos = [yr, xc]
```

Comment:
Just a minor nitpick: When `yr` and `xc` are the coordinates of the center of the selected side band, why aren't they called `yc` and `xc`(c for center, I assume)? Or what does the r in `yr` mean? Also, the convention is to use a tuple instead of a list for a shape (which I assume `sb_pos` should be.
Furthermore, you can directly assign:
```python
sb_pos = np.unravel_index(eh_hw_fft_half.argmax(), eh_hw_fft_half.shape)
```
and later use `sb_pos[0]` and `sb_pos[1]` instead of `yc` and `xc`